Expensify/expensify-common
# Fix new lines when replacing <blockquote> and <br />

This is part of a group of changes to preserving new lines when converting from `message[0].html` using `ExpensiMark.htmlToText`. This should improve:
- Copy message content to clipboard (Copy text using the clipboard button, and past in text editor: the line breaks are missing. Related issue: https://github.com/Expensify/App/issues/10262)
- It is going to be useful for showing only the first line of a message as in the LHN (currently `<blockquote>` cause buggy spacing)


### Fixed Issues
$ https://github.com/Expensify/App/issues/12894 (partially)
https://github.com/Expensify/App/issues/11494 (partially)

# Tests

Make sure unit test run (`npm run test`)

1. Go to a chat
2. Go offline
3. Compose a message by:
    1. Add an emoji
    2. Add two or more empty lines
    4. Add an emoji
    ![image](https://user-images.githubusercontent.com/87341702/203211230-dd25e00b-a4b8-4b58-82e0-e17852d45d85.png)
4. Send the message
5. Find report action in Onyx and make sure that the line breaks appear as `\n` in `message[0].text`
    ![image](https://user-images.githubusercontent.com/87341702/204638681-4e99c2ac-c1b0-4456-98c1-203b6c42c5a4.png)
7. Compose another message with the content:
```
before quotes
> quoted text
after quotes
```
5. Find report action in Onyx and make sure that the `message[0].text` contains `before quotes\nquoted text\nafter quotes`. This is because `<blockquote>` are block elements and they should cause line breaks in the content.
    ![image](https://user-images.githubusercontent.com/87341702/204638189-e595f93b-1640-44ac-a78c-464507e24588.png)






# QA


1. Go to a chat
2. Go offline
3. Compose a message by:
    1. Add an emoji
    2. Add two or more empty lines
    4. Add an emoji
    ![image](https://user-images.githubusercontent.com/87341702/203211230-dd25e00b-a4b8-4b58-82e0-e17852d45d85.png)
4. Send the message
5. Find report action in Onyx and make sure that the line breaks appear as `\n` in `message[0].text`
    ![image](https://user-images.githubusercontent.com/87341702/204638681-4e99c2ac-c1b0-4456-98c1-203b6c42c5a4.png)
7. Compose another message with the content:
```
before quotes
> quoted text
after quotes
```
5. Find report action in Onyx and make sure that the `message[0].text` contains `before quotes\nquoted text\nafter quotes`. This is because `<blockquote>` are block elements and they should cause line breaks in the content.
    ![image](https://user-images.githubusercontent.com/87341702/204638189-e595f93b-1640-44ac-a78c-464507e24588.png)
